kernel/timeout.c

```diff
@@ -326,6 +326,25 @@ k_timeout_t sys_timepoint_timeout(k_timepoint_t timepoint)
 	return K_TICKS(remaining);
 }
 
+bool sys_timepoint_lessthan(k_timepoint_t a, k_timepoint_t b)
+{
+	return a.tick < b.tick;
+}
+
+bool sys_timeout_lessthan(k_timeout_t a, k_timeout_t b)
```

Comment:
This is getting back into the weeds of "math on timeouts" which timepoint (like sys_timeout_end_calc() before it) is supposed to be supplanting.

A timeout is a command, not a time.  It's a way to tell the kernel how long in the future something is going to be, but the actual time value it represents won't be known until you give it to the kernel.  "Now" at the point of a `sys_timeout_lessthan()` call is not the same "now" as e.g. the call to k_sleep() it will eventually control.

The idea is that, if you really really need to do math on these things (and if you really do, you almost certainly want to be using absolute timeouts), you instantiate a timepoint from your timeout immediately before passing it to the kernel, and use that.